### PR TITLE
br: add lock file

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
Fix https://github.com/pingcap/br/issues/328
Add a lock file at beginning, to make sure other backup task won't be able to start backup.

### What is changed and how it works?
Add a lock file when storage created.

### Check List 

Tests 

 - Integration test

### Release Note

 - Fix the issue that fast fail when multiple backup task start backup on same path.

### CODE REVIEW

**Comment by 3pointer** [comment]
Maybe we should move `backupmeta` check, since it not necessary from now.

**Comment by kennytm** [comment]
cc #151 the previous attempt.

**Comment by kennytm** [comment]
LGTM but the file name sounds odd.

Unlike a typical lock file, we do not delete `backup.lock` after backup has completed.

**Comment by 3pointer** [comment]
> LGTM but the file name sounds odd.
> 
> Unlike a typical lock file, we do not delete `backup.lock` after backup has completed.

yes, if we want to delete file we need add `delete` method for external storage, I wonder is it necessary, because it does make sense even one backup finished, other backup tasks still won't be able to use same path. maybe change lock file to another name is better, do you have any advise?

File: pkg/backup/client.go [hunk 1]
```diff
@@ -122,6 +122,11 @@ func (bc *Client) GetTS(ctx context.Context, duration time.Duration, ts uint64)
 	return backupTS, nil
 }
 
+// SetLockFile set write lock file.
+func (bc *Client) SetLockFile(ctx context.Context) error {
+	return bc.storage.Write(ctx, utils.LockFile, []byte{})
```
Comment (overvenus) [inline pkg/backup/client.go:127]: Could you write some warnings to the file? Eg. DO NOT DELETE and explain what the file does.

**Pull request merged**